### ISSUE TITLE
Ability to hide trees

### ISSUE THREAD

**Issue by Razip** [OpenRCT2/OpenLoco#116]

**Comment by duncanspumpkin** [comment]
Flesh out the issue more. Provide screen shots of why the existing filters are not good enough.

**Comment by auge8472** [comment]
I hope that I got the purpose of Razips issue. I'll use screenshots from OpenTTD because I haven't installed Locomotion or OpenLoco at the moment.

If you use the hide-function in Locomotion, the hidden items (houses, trees, etc.) are rendered as a shadow (see screenshot 1 with a similar look in OpenTTD). This is IMHO not optimal. I can't see everything I want, especially when several shadows overlap, and, when scrolling, it has a unfavourable influence to the rendering performance.

screenshot 1:

![OpenTTD, hidden trees with shadows](https://user-images.githubusercontent.com/1734660/36350999-ad202512-14a2-11e8-9ddd-a21a7b53b27d.png)

In OpenTTD on the other hand one can toggle between hiding the items with shadow (to show, where they are placed) and hiding them completely (for a clearer view, see screenshot 2, OpenTTD with completely hidden trees).

screenshot 2:

![OpenTTD, hidden trees without shadows](https://user-images.githubusercontent.com/1734660/36351018-04748718-14a3-11e8-861c-46c896bc0dca.png)

This issue is not only a problem for trees. So it's preferable to have a switch for all items, that can be hidden. Either *one* switch *for all* shadows *or one per item class* to hide like in OpenTTD. See therefore the slim buttons below the icons at the bottom of the transparency window in the screenshots)

**Comment by Razip** [comment]
![2018-02-18_194246](https://user-images.githubusercontent.com/6756114/36353136-33186a4e-14e4-11e8-9918-a00619e234eb.jpg)
1) The shadows may distract or bother people, so I think it'd be cool if we had an option to turn them off.
2) The filter hides only a portion of displayed buildings, not all them which can bother too.

**Comment by jgottula** [comment]
I agree that the thing-hiding functionality in Locomotion needs to be be expanded upon substantially.

- The game only really gives you two options for hiding things: it just dumps everything into the "all trees and buildings and whatnot" or "all roads and tracks of every kind" buckets; and if you want to hide more specific sub-types of things, you're just out of luck. I mean, hell, RCT gave you more granular hiding options in its view menu.
- The inability to have things be completely invisible is pretty bad. When there are many translucent things on-screen, the rendering indeed slows down horrendously, at least with the vanilla drawing code. (Particluarly at higher resolutions, and particularly when there are more things on-screen due to being zoomed out.) Also, in densely forested areas, the translucency of the many trees combines together and the translucency becomes so dark that it's more-or-less impossible to see details of the landscape behind them because of all the darkness. (The workaround for this is generally to get out the bulldozer and cut down all the trees so you can see what you're doing, which tends to make the townsfolk grumpy...)
- The inability to do anything other than hide on the bottom 50% of the screen is annoying at times. Only hiding on the bottom half of the screen **does** make some sense in many ways; however, the fact that Chris only gave us that one type of functionality with no alternatives is a little baffling and can be constraining.

----

Personally, I think it'd be cool if we could eventually implement all of the following:

- More granular hiding toggles in the view menu. Examples of groups of things that would ostensibly be individual options for hiding:
  - Trees (and other scenary-ish things like fences and whatever)
  - City buildings
  - Industy buildings
  - Roads
  - Rails
  - Tram tracks
  - (Perhaps) Only *enemy-company-owned* roads/rails/tracks, or only *city-owned* roads, etc.
  - (I'm probably forgetting some other classes of things here that would also be applicable...)
- Ability to choose between translucency and complete invisibility
- Ability to choose to have things be translucent/invisible across *the entire screen*, rather than merely the bottom 50% (or perhaps a selectable percentage ratio slider, or something)

Many of the finer-grained aspects here (translucency vs invisibility for each thing; how much of the screen to make translucent/invisible for each thing; that kind of stuff) would probably be too cumbersome to integrate all into the view menu at once (depending on how we'd do the menu: checkboxes, or submenus, or whatever); so perhaps in the future we might invent a new options tab for "visibility options" and then allow these sorts of preferences to be set in a more static manner via that.

Ideally it'd be nice if the translucency/invisibility and how-much-of-the-screen-to-hide parameters could be set independently for each different type of thing that's separately hidable in the view menu (i.e. you could set up "hide trees" so that they'll be completely invisible across the entire screen; and then also have "hide industry buildings" set up so that they're just translucent on the bottom 50% of the screen; and so forth). I dunno whether that'd make for too complicated of an options UI. Seems like it wouldn't be too horrible if we did a sort of table layout: with a row for each thing type; and then a column with checkboxes/dropdowns for translucent/invisible; and another column with a checkbox or dropdown or slider for how-much-of-the-screen-to-hide-things-on.

Maybe we could even add a convenience menu item at the bottom of the view menu specifically for the visibility options tab, similar to how the sound menu has its own menu item specifically for the sound options tab.

----

Here's a rough UI mock-up for the options UI as I envisioned it in my head, just to give a visual representation in case my textual description didn't get the idea across fully.

(Note: I only put a few rows in the mockup; also the dropdown could perhaps be a checkbox; also the slider could potentially be a checkbox or dropbown; etc etc etc.)

![openloco_visibility_options_mockup](https://user-images.githubusercontent.com/822207/36356559-a7a4d01e-14a8-11e8-8cc1-13ef8ff95e16.png)

(Paint.NET file that I used to make that mockup: [OpenLoco_Visibility_Options_Mockup.pdn.gz](https://github.com/OpenRCT2/OpenLoco/files/1735103/OpenLoco_Visibility_Options_Mockup.pdn.gz))

**Comment by jgottula** [comment]
> (I'm probably forgetting some other classes of things here that would also be applicable...)

Aaaaaaaand, I just remembered: being able to hide vehicles (potentially by type) might also be useful.

**Comment by jgottula** [comment]
> Also, in densely forested areas, the translucency of the many trees combines together and the translucency becomes so dark that it's more-or-less impossible to see details of the landscape behind them because of all the darkness.

Here are some screenshots demonstrating this. (Admittedly these were contrived in the scenario editor, but I've come across some stuff this bad in real scenarios. Maybe not as bad as the third image, but definitely as bad as the first two.)

![screenshot83](https://user-images.githubusercontent.com/822207/36356711-7ba46806-14aa-11e8-89fb-3b67db75b61f.png)
![screenshot84](https://user-images.githubusercontent.com/822207/36356713-7d7cc22c-14aa-11e8-9348-85bdbd6a96d9.png)
![screenshot85](https://user-images.githubusercontent.com/822207/36356715-7f322ea4-14aa-11e8-9f52-bf22fa9613d3.png)

**Comment by Razip** [comment]
@jgottula, I agree with you, but I have a question: why would anyone ever need to hide, let's say, 50% of the displayed trees, 25% of the displayed houses and 75% of the industries? I think that most people would need 100% of the stuff to be hidden.

**Comment by auge8472** [comment]
Like @Razip said, it is not very common to distinguish between the dimensions of items to hide. IMHO it is enogh to be able to hide items of class `$x` and to hide them translucent or hide them comlete. The screenshots from OpenTTD showing the way, it's done there. An icon button per item class and below it a further button to toggle between translucency and invisibility.

`TODO` for @auge8472 (me): install Locomotion and OpenLoco!

**Comment by jgottula** [comment]
> @jgottula, I agree with you, but I have a question: why would anyone ever need to hide, let's say, 50% of the displayed trees, 25% of the displayed houses and 75% of the industries? I think that most people would need 100% of the stuff to be hidden.

@Razip I don't really know; I just figured that, since having only-foreground-objects-hidden *can* be a situationally useful feature, it might as well become a fully configurable thing. Because maybe in certain situations, still having certain types of objects visible across a particular percentage of the vertical range of the screen might be handy.

But I fully admit that going "full slider" with that aspect of it is almost certainly a bit too far. 😝

**Comment by Razip** [comment]
@jgottula, then let 100% of each category of things be semi-transparent or fully transparent by default.

**Comment by jgottula** [comment]
It's just a rough UI mock-up; don't make too much of it.

**Comment by Razip** [comment]
There's one annoying thing about the original filter: you can't remove objects like bus stops behind the semi-transparent obstacles until you destroy all of the latter.

**Comment by YetiWiz** [comment]
> There's one annoying thing about the original filter: you can't remove objects like bus stops behind the semi-transparent obstacles until you destroy all of the latter.

Exactly, it makes building subways hell, because you end up destroying half the city just to select the damn underground track.

**Comment by IntelOrca** [comment]
The cut-away view introduced in OpenRCT2 would be useful here.

**Comment by AaronVanGeffen** [comment]
The granular toggles have been implemented by #2231, which was just merged and will be in the next release. An option to disable the 'see-through' blur will likely be added in the near future. Leaving this open to track that.
